project/FrgaalJavaCompiler.scala

```diff
@@ -63,6 +64,58 @@ object FrgaalJavaCompiler {
     val sources = sources0 map {
       case x: PathBasedFile => x.toPath.toAbsolutePath.toString
     }
+    val out = output.getSingleOutputAsPath().get()
+    val shared = sources0.fold(out)((a, b) => {
+      var ap = a match {
+        case p: PathBasedFile => p.toPath
+        case p: Path => p
+      }
+      val bp = b match {
+        case p: PathBasedFile => p.toPath
+        case p: Path => p
+      }
+
+      var i = 0
+      while (i < Math.min(ap.getNameCount(), bp.getNameCount()) && ap.getName(i) == bp.getName(i)) {
+        i += 1;
+      }
+
+      while (ap.getNameCount() > i) {
+        ap = ap.getParent()
+      }
+      ap
+    }).asInstanceOf[Path];
+
+    if (shared.toFile().exists()) {
+      val ensoMarker = new File(shared.toFile(), ".enso-sources")
+      val ensoConfig = new File(shared.toFile(), ".enso-sources-" + out.getFileName().toString())
+      val p = new java.util.Properties()
```

Comment:
Not a fan of single letter vars :)

Reply:
Renamed in [3de9908](https://github.com/enso-org/enso/pull/3810/commits/3de99088f149fa8f6b64e5426db4d80c0278e21c)